tools/scripts/deploy.sh

```diff
@@ -0,0 +1,132 @@
+            --proxy-user)         PROXY_USER="$2"; shift; shift ;;
+            --proxy-pass)         PROXY_PASS="$2"; shift; shift ;;
+            --target-ip)          TARGET_IP="$2"; shift; shift ;;
+            --target-user)        TARGET_USER="$2"; shift; shift ;;
+            --target-pass)        TARGET_PASS="$2"; shift; shift ;;
+            -- ) shift; break ;;
+            * ) err "unsupported argument $1"; usage; exit 1 ;;
+        esac
+    done
+
+    echo "Start"
+    tmpdir=$(mktemp -d)
+    mkdir -p "${tmpdir}"/opt/prplmesh
+    mkdir -p "${tmpdir}"/usr
+
+    cp -r "${BUILD_DIR}"/install/bin "${tmpdir}"/opt/prplmesh
+    cp -r "${BUILD_DIR}"/install/config "${tmpdir}"/opt/prplmesh
+    cp -r "${BUILD_DIR}"/install/lib "${tmpdir}"/usr/
+    cp -r "${BUILD_DIR}"/install/scripts "${tmpdir}"/opt/prplmesh
+    cp -r "${BUILD_DIR}"/install/share "${tmpdir}"/opt/prplmesh
+
+    dbg "create prplmesh.tar.gz..."
+    tar czvf "${BUILD_DIR}"/prplmesh.tar.gz -C "${tmpdir}"/ .
+    rm -rf "${tmpdir}"
+
+    [ "$DEPLOY" = "true" ] && {
+
+        echo "local md5sum"
+        run md5sum "${BUILD_DIR}"/prplmesh.tar.gz
+
+        if [ -n "$PROXY_IP" ] && [ -n "$PROXY_USER" ] && [ -n "$PROXY_PASS" ]; then
+
+            echo "scp ${BUILD_DIR}/prplmesh.tar.gz to ${PROXY_USER}@${PROXY_IP}"
+            run sshpass -p "${PROXY_PASS}" scp "${SSH_OPTIONS}" "${BUILD_DIR}"/prplmesh.tar.gz "${PROXY_USER}"@"${PROXY_IP}":/home/"${PROXY_USER}"/
+
+            echo "scp ${PROXY_USER}@${PROXY_IP}:/home/${PROXY_USER} to ${TARGET_USER}@${TARGET_IP}:/tmp"
+            run sshpass -p "${PROXY_PASS}" ssh "${SSH_OPTIONS}" "${PROXY_USER}"@"${PROXY_IP}" "sshpass -p ${TARGET_PASS} scp ${SSH_OPTIONS} /home/${PROXY_USER}/prplmesh.tar.gz ${TARGET_USER}@${TARGET_IP}:/tmp/"
+
+            echo "remote md5sum"
+            run sshpass -p "${PROXY_PASS}" ssh "${SSH_OPTIONS}" "${PROXY_USER}"@"${PROXY_IP}" "sshpass -p ${TARGET_PASS} ssh ${SSH_OPTIONS} ${TARGET_USER}@${TARGET_IP} md5sum /tmp/prplmesh.tar.gz"
+
+            echo "backaup /opt/prplmesh/beerocks_agent.conf on target"
+            run sshpass -p "${PROXY_PASS}" ssh "${SSH_OPTIONS}" "${PROXY_USER}"@"${PROXY_IP}" "sshpass -p ${TARGET_PASS} ssh ${SSH_OPTIONS} ${TARGET_USER}@${TARGET_IP} cp /opt/prplmesh/config/beerocks_agent.conf /tmp/"
+
+            echo "untar /tmp/prplmesh.tar.gz on target"
+            run sshpass -p "${PROXY_PASS}" ssh "${SSH_OPTIONS}" "${PROXY_USER}"@"${PROXY_IP}" "sshpass -p ${TARGET_PASS} ssh ${SSH_OPTIONS} ${TARGET_USER}@${TARGET_IP} tar -C / -xzvf /tmp/prplmesh.tar.gz"
+
+            echo "restore /opt/prplmesh/beerocks_agent.conf on target"
+            run sshpass -p "${PROXY_PASS}" ssh "${SSH_OPTIONS}" "${PROXY_USER}"@"${PROXY_IP}" "sshpass -p ${TARGET_PASS} ssh ${SSH_OPTIONS} ${TARGET_USER}@${TARGET_IP} cp /tmp/beerocks_agent.conf /opt/prplmesh/config/"
+    
+        elif [ -n "$TARGET_IP" ] && [ -n "$TARGET_USER" ] && [ -n "$TARGET_PASS" ]; then
```

Comment:
What if you don't use a password for the target (and use ssh keys instead)? It looks like the script won't do anything if you don't provide a password.

Reply:
Completely removed the password and use of sshpass.
The user is expected to set up ssh keys via `ssh-keygen; ssh-copy-id <proxy>`, otherwise he will be prompt for entering the proxy password.